game-core/src/main/java/games/strategy/engine/lobby/moderator/toolbox/CreateNewApiKeyActions.java

```diff
@@ -16,23 +18,20 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 class CreateNewApiKeyActions {
 
-  static boolean registerApiKey(
-      final URI serverUri,
-      final String apiKey,
-      final String newPassword) {
+  static boolean registerApiKey(final URI serverUri, final ApiKeyPassword apiKeyPassword) {
 
-    final ModeratorToolboxClient toolboxClient =
-        ModeratorToolboxClient.newClient(serverUri, newPassword);
+    final ToolboxRegisterNewKeyClient registerNewKeyClient =
+        ToolboxRegisterNewKeyClient.newClient(serverUri);
 
-    final RegisterApiKeyResult result = toolboxClient.registerNewKey(apiKey);
+    final RegisterApiKeyResult result = registerNewKeyClient.registerNewKey(apiKeyPassword);
 
-    return result.getNewApiKey()
+    return Optional.ofNullable(result.getNewApiKey())
```

Comment:
@RoiEXLab API was updated to return non-optional values, that was important for JSON encoding to not encode optional objects into the JSON